presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -645,6 +656,17 @@ static OptionalInt getBucketNumber(String name)
         return OptionalInt.empty();
     }
 
+    @VisibleForTesting
+    static Optional<String> getAttemptId(String bucketFilename)
+    {
+        Matcher matcher = BUCKET_WITH_OPTIONAL_ATTEMPT_ID_PATTERN.matcher(bucketFilename);
+        if (matcher.matches() && matcher.groupCount() == 2 && matcher.group(2) != null) {
```

Comment:
checking `matcher.groupCount() == 2` is wrong, since the group count is statically defined in the pattern

```suggestion
        if (matcher.matches() && matcher.group(2) != null) {
```